### PR TITLE
[SYCL] Add test for AllocaCommand cleanup

### PR DESCRIPTION
**Pull request by abuyukku**

### CODE REVIEW

File: SYCL/Scheduler/AllocaCleanup.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    });
+    q.wait();
+
+    std::vector<sycl::buffer<RealType, 1>> sub_helper; // props
+    for (int i = 0; i < p; i++) {
+        sub_helper.push_back(sycl::buffer<RealType, 1>(helper_buf, i*n, n)); 
+    }
+
+    auto policy = oneapi::dpl::execution::make_device_policy(q);
+    auto w_begin = oneapi::dpl::begin(w_buf);
```
Comment (romanovvlad) [inline SYCL/Scheduler/AllocaCleanup.cpp:39]: I believe it's better to avoid dependencies on non SYCL APIs in these tests.
Tagging @vladimirlaz

Comment (vladimirlaz) [reply SYCL/Scheduler/AllocaCleanup.cpp:39]: It is better to avoid extra dependencies in these tests? Is it possible to remove the use of DPL?

Comment (abuyukku) [reply SYCL/Scheduler/AllocaCleanup.cpp:39]: Added is the reproducer for the sporadic segfaults fixed in https://github.com/intel/llvm/pull/4573. I agree that having dependencies to other moving components in SYCL tests is not a good idea, however this was the only reproducer I have at hand. I'll check to see if the issue reproduces even after removing the DPL dependencies.

**Comment by abuyukku** [comment]
Closing this PR. intel/llvm#4573 adds in a unit test for the failing scenario fixed there.

**Pull request closed without merging**